.pipelines/.vsts-vhd-builder.yaml

```diff
@@ -130,30 +132,15 @@ stages:
             artifactName: 'vhd-image-bom'
             targetPath: 'image-bom.json'
           displayName: publish container image list
-        - bash: |
-            echo MODE=$(MODE) && \
-            OS_DISK_SAS="$(cat packer-output | grep "OSDiskUriReadOnlySas:" | cut -d " " -f 2)" && \
-            docker run --rm \
-            -v ${PWD}:/go/src/github.com/Azure/AgentBaker \
-            -w /go/src/github.com/Azure/AgentBaker \
-            -e CLIENT_ID=${CLIENT_ID} \
-            -e CLIENT_SECRET="$(CLIENT_SECRET)" \
-            -e TENANT_ID=${TENANT_ID} \
-            -e SUBSCRIPTION_ID="${SUBSCRIPTION_ID}" \
-            -e CLASSIC_BLOB=${CLASSIC_BLOB} \
-            -e CLASSIC_SAS_TOKEN="$(SAS_TOKEN)" \
-            -e OS_DISK_SAS=${OS_DISK_SAS} \
-            ${CONTAINER_IMAGE} make -f packer.mk az-copy
-          displayName: Copying resource to Classic Storage Account
-          condition: and(succeeded(), eq(variables.DRY_RUN, 'False'), eq(variables['MODE'], 'default'))
+          condition: and(succeeded(), eq(variables.DRY_RUN, 'False'), eq(variables['MODE'], 'linuxVhdMode'))
         - bash: |
             echo MODE=$(MODE) && \
             PKR_RG_NAME="$(cat packer-output | grep "ResourceGroupName" | cut -d "'" -f 2 | head -1)" && \
             SA_NAME="$(cat packer-output | grep "storage name:" | cut -d " " -f 3)" && \
             gen2_captured_sig_version="$(cat vhdbuilder/packer/settings.json | grep "gen2_captured_sig_version" | awk -F':' '{print $2}' | awk -F'"' '{print $2}')" && \
             IMPORTED_IMAGE_NAME="$(cat vhdbuilder/packer/settings.json | grep "imported_image_name" | awk -F':' '{print $2}' | awk -F'"' '{print $2}')" && \
             ARM64_OS_DISK_SNAPSHOT_NAME="$(cat vhdbuilder/packer/settings.json | grep "arm64_os_disk_snapshot_name" | awk -F':' '{print $2}' | awk -F'"' '{print $2}')" && \
-            if [[ $(MODE) == "gen2Mode" ]]; then IMAGE_NAME="${gen2_captured_sig_version}"; else IMAGE_NAME=$SIG_IMAGE_NAME; fi && \
+            if [[ "${HYPERV_GENERATION,,}" == "v2" ]]; then IMAGE_NAME="${gen2_captured_sig_version}"; else IMAGE_NAME=$SIG_IMAGE_NAME; fi && \
```

Comment:
typo?

Reply:
instead of checking MODE we now check HYPERV_GENERATION, don't think I see a typo here? unless you're saying that we shouldn't be using HYPERV_GENERATION here

Reply:
granted I'm thinking about how we can change this logic altogether since we'll always need a unique captured sig version at this point

Reply:
yeah, so I'm gonna change this such that we always get the image name from the captured_sig_version since we'll no longer have a case where packer outputs a disk URI

Reply:
sorry to interrupt here. are you suggesting packer will always output a managed sig image, and so the image will have to be snapshot into an OS disk blob before publishing?

Reply:
these changes will have it such that packer always outputs a managed image and publishes it to a SIG version for Linux-based VHDs - this means that we will always be converting a SIG image version to classic storage account blob, which will then be forwarded on to the appropriate release pipelines

I've made sure to keep the windows build definitions/modes the same with these changes however